pkg/datapath/loader/netlink.go

```diff
@@ -329,3 +332,58 @@ func SetupBaseDevice(mtu int) (netlink.Link, netlink.Link, error) {
 
 	return linkHost, linkNet, nil
 }
+
+// reloadIPSecOnLinkChanges subscribes to link changes to detect newly added devices
+// and reinitializes IPsec on changes. Only in effect for ENI mode in which we expect
+// new devices at runtime.
+func (l *Loader) reloadIPSecOnLinkChanges() {
+	// settleDuration is the amount of time to wait for further link updates
+	// before proceeding with reinitialization. This avoids back-to-back
+	// reinitialization when multiple link changes are made at once.
+	const settleDuration = 1 * time.Second
+
+	if !option.Config.EnableIPSec || option.Config.IPAM != ipamOption.IPAMENI {
```

Comment:
side-note: We don't support AWS CNI chaining with IPSec at the moment, so this is fine for now. But long-term, it might make sense to enable this feature independent of the IPAM mode, since I could imagine that there might be other setups (besides ENI mode) where pod egress/ingress interfaces may be added at run-time.

I wouldn't be surprised if alibabacloud or azure modes suffer from similar issues, but I'm not familiar enough with them to judge this.

Reply:
Longer-term I want to get rid of the bpf_network. bpf_host can already do the same job.

> I wouldn't be surprised if alibabacloud or azure modes suffer from similar issues, but I'm not familiar enough with them to judge this.

Does Azure have any sort of interfaces like ENIs that can be added at runtime?

Reply:
I'm not 100% sure, but looking at the `NodeOperations` of AlibabaCloud and Azure, it looks like AlibabaCloud has a method to add new interfaces, but Azure does not:

https://github.com/cilium/cilium/blob/7e5bf218d963d2800ab6351373b13540faba394d/pkg/azure/ipam/node.go#L133-L137